### PR TITLE
feat: Add shared element transition

### PR DESCRIPTION
**Pull request by anhanh11001**
Fixes: #1686

- Add shared element transition between event detail and event list.
- Remove unused Scope file.

Screenshots for the change:
ezgif-2-5e8abfb58111  ezgif-2-069709e4221f  ezgif-2-c012b715d128

### CODE REVIEW

**Comment by aggarwalpulkit596** [comment]
@anhanh11001 please refer this you willl find solution for return transition #1443

**Comment by aggarwalpulkit596** [comment]
@anhanh11001 if you're not working on this should I check #1443 for the bug that caused for it to be reverted

**Comment by anhanh11001** [comment]
Actually, I have finished them (I think so), I haven't uploaded the latest version because it slows down the app and at first I thought it was some problem in this PR but then I figured it out it was #1705. So I want to finish 1705 first before actually testing the performance of this PR. So maybe if you can also take a look at #1705 and also send a PR about it, that would be nice

**Comment by aggarwalpulkit596** [comment]
@anhanh11001 sure will take a look and can you take a look at #1477 to make sure these bugs are not happening again

**Comment by anhanh11001** [comment]
@aggarwalpulkit596 Yes I will, if I still have problems I will tell you

File: app/src/main/java/org/fossasia/openevent/general/event/EventsFragment.kt [hunk 1]
```diff
@@ -157,10 +161,26 @@ class EventsFragment : Fragment(), ScrollToTop {
...
+                val itemEventViewHolder = rootView.eventsRecycler.findViewHolderForAdapterPosition(itemPosition)
+                itemEventViewHolder?.let {
+                    if (itemEventViewHolder is EventViewHolder) {
+                        extras = FragmentNavigatorExtras(
+                            itemEventViewHolder.itemView.eventImage to "eventDetailImage")
+                    }
+                }
+                if (extras != null) {
+                    findNavController(view).navigate(EventsFragmentDirections.actionEventsToEventsDetail(eventID),
+                        extras as Navigator.Extras)
```
[the remark this answers is not in the document]
Comment (liveHarshit) [reply app/src/main/java/org/fossasia/openevent/general/event/EventsFragment.kt]: Can you avoid casting?

**Comment by anhanh11001** [comment]
updated

**Comment by liveHarshit** [comment]
@anhanh11001 Have you tested it for search results? Search is not working and the app is crashing on going back.

**Comment by aggarwalpulkit596** [comment]
@liveHarshit it is same error which was in my implementation

**Comment by aggarwalpulkit596** [comment]
There is some issue with the navigation architecture i guess

**Comment by liveHarshit** [comment]
> @liveHarshit it is same error which was in my implementation

Likes and events are working fine. Heare the issue is with search fragment

**Comment by aggarwalpulkit596** [comment]
check #1477

**Comment by liveHarshit** [comment]
You can compare if the method of implementation is same.

**Comment by anhanh11001** [comment]
@liveHarshit can you tell me how to reproduce it, because I have tested and it is working fine in my device

**Comment by liveHarshit** [comment]
On my device search is not redirecting to search fragment, only title changes and after coming back app crashes.

**Comment by liveHarshit** [comment]
Let me share logs

**Comment by liveHarshit** [comment]
```
E/EventViewHolder$bind$$inlined$let$lambda: com.squareup.picasso.NetworkRequestHandler$ResponseException: HTTP 404
        at com.squareup.picasso.NetworkRequestHandler.load(NetworkRequestHandler.java:51)
        at com.squareup.picasso.BitmapHunter.hunt(BitmapHunter.java:219)
        at com.squareup.picasso.BitmapHunter.run(BitmapHunter.java:175)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:428)
        at java.util.concurrent.FutureTask.run(FutureTask.java:237)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607)
        at java.lang.Thread.run(Thread.java:760)
        at com.squareup.picasso.Utils$PicassoThread.run(Utils.java:354)
```

```
E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.eventyay.attendee, PID: 10831
    java.lang.NullPointerException: Attempt to invoke virtual method 'boolean androidx.fragment.app.FragmentManagerImpl.isPrimaryNavigation(androidx.fragment.app.Fragment)' on a null object reference
        at androidx.fragment.app.Fragment.performPrimaryNavigationFragmentChanged(Fragment.java:2710)
        at androidx.fragment.app.FragmentManagerImpl.dispatchParentPrimaryNavigationFragmentChanged(FragmentManagerImpl.java:2795)
        at androidx.fragment.app.FragmentManagerImpl.setPrimaryNavigationFragment(FragmentManagerImpl.java:2790)
        at androidx.fragment.app.BackStackRecord.executeOps(BackStackRecord.java:422)
        at androidx.fragment.app.FragmentManagerImpl.completeExecute(FragmentManagerImpl.java:1979)
        at androidx.fragment.app.FragmentManagerImpl$StartEnterTransitionListener.completeTransaction(FragmentManagerImpl.java:3371)
        at androidx.fragment.app.FragmentManagerImpl.executePostponedTransaction(FragmentManagerImpl.java:1759)
        at androidx.fragment.app.FragmentManagerImpl.removeRedundantOperationsAndExecute(FragmentManagerImpl.java:1790)
        at androidx.fragment.app.FragmentManagerImpl.execPendingActions(FragmentManagerImpl.java:1717)
        at androidx.fragment.app.FragmentManagerImpl$2.run(FragmentManagerImpl.java:150)
        at android.os.Handler.handleCallback(Handler.java:754)
        at android.os.Handler.dispatchMessage(Handler.java:95)
```

**Comment by anhanh11001** [comment]
@liveHarshit I think I know the problem now, it is not the same as #1477, fixing now

**Pull request merged**